Modules/StudyProgramme/classes/class.ilObjStudyProgramme.php

```diff
@@ -216,16 +216,14 @@ protected function deleteSettings() : void
      *
      * @throws ilException
      */
-    protected function deleteAssignments() : void
+    protected function deleteAssignmentsAndProgresses() : void
     {
-        foreach ($this->getAssignments() as $ass) {
-            $progresses = $this->getProgressForAssignment($ass->getId());
-            foreach ($progresses as $progress) {
-                $progress->delete();
-            }
-
-            $this->assignment_repository->delete($ass);
-        }
+        $assignment_repository = $this->getAssignmentRepository();
+        $assignment_repository->deleteAllAssignmentsForProgrammeId($this->getId());
+        $orphan_condition_field = $assignment_repository->getTableAndFieldOfAssignmentIds();
+        $progress_repository = $this->getProgressRepository();
+        $progress_repository->deleteProgressesFor($this->getId());
+        $progress_repository->deleteAllOrphanedProgresses(...$orphan_condition_field);
```

Comment:
This is another indicator for the impression that assignment and progress ought to have one common repository...